src/router/parametersPreProcess.routerPlugin.js

```diff
@@ -0,0 +1,86 @@
+import { MAP_VIEWS } from '@/router/viewNames'
+import { WEBMERCATOR } from '@/utils/coordinates/coordinateSystems'
+import log from '@/utils/logging'
+import { isNumber } from '@/utils/numberUtils'
+
+/**
+ * @param {String} param_name
+ * @param {any} param_value
+ * @param {Object} newQuery The modified query for the router
+ */
+const preProcessParam = (param_name, param_value, query) => {
+    switch (param_name) {
+        case '3d':
+            // if the 3d flag is set, we need to ensure that we are in the webmercator projection
+            if (
+                ((typeof param_value === 'boolean' && param_value) ||
+                    param_value === 'true' ||
+                    param_value === null) &&
+                (!isNumber(query.sr) || Number(query.sr) !== WEBMERCATOR.epsgNumber)
+            ) {
+                log.info(
+                    `[URL Param pre process] 3d parameter found. Ensuring spatial reference system is set to web mercator`
+                )
+                query['sr'] = WEBMERCATOR.epsgNumber
+            }
+            break
+        default:
+            break
+    }
+}
+
+const handleParams = async (query) => {
+    // if so, we transfer all old param (stored before vue-router's /#) and transfer them to the MapView
+    // we will also transform legacy zoom level here (see comment below)
+    const newQuery = structuredClone(query)
```

Comment:
I'm not sure it is a very good idea to use such new web api released only 2 years ago, this might not work for a lot of peoples, @pakb what do you think ? It might make more sense to use the lodash deep clone function